{{cookiecutter.repostory_name}}/pyproject.toml

```diff
@@ -90,14 +87,15 @@ line-length = 120
 [tool.ruff.lint]
 # TODO add D
 select = [
-    "E", "F", "I", "UP",
+    "E", "F", "I", "UP", "S",
     "TCH005",
 ]
 # TODO: remove E501 once docstrings are formatted
 ignore = [
     "D100", "D105", "D107", "D200", "D202", "D203", "D205", "D212", "D400", "D401", "D415",
     "D101", "D102","D103", "D104", # TODO remove once we have docstring for all public methods
     "E501", # TODO: remove E501 once docstrings are formatted
+    "S101", "S105", "S106", "S108", "S311", "S602", "S603", "S607",
```

Comment:
Well, this sounds reasonable conceptually, but the problem is that all of these rules (except S108) have lots of false positives in one of our projects. My concern is that the hassle of dealing with them on a daily basis might outweigh the actual benefits they provide.
At least when it comes to S6**, don't you think that if you're making subprocess calls, you probably know what you're doing?
I also don't see a real problem in using assert in a production code sometimes. Off course an explicit exception should be raised most of the time, but if it's something that should never be happening, `assert` is okay for me. Maybe I'm wrong here.
I suggest ignoring at least: S101, S602 and S603. What do you think?

Reply:
I guess S101 can be legit used in prod code - for example to tell mypy to chillout

S602, S603, welp I guess these are not as important as you say - they require special care anyhow.

But everything else - if there is some special need project already - it just needs to extend its list of ignored rules on its own. AFAIK the aformentioned project is not really being kept up to date to template so update by hand will be necessary there anyhow.

Hm, btw, now that I look at S108 - i guess we can always ignore that one... if attacked can poison PATH then you probably are already screwed and figuring out the absolute path in virtualenv context seems PITA.

Reply:
`S603` has too many false positives. There's even a [corresponding issue](https://github.com/astral-sh/ruff/issues/4045) pinned in ruff docs. `S602` can be useful though, agree, removed it from the ignored list
`S607` is also quite cumbersome to deal with, especially when one runs commands on different linux systems, absolute path can be hard to know in advance. This gives 4 errors even in a freshly generated cookiecutter project because of commands in `vultr-scripts`. Added to ignore list
Agree about `S108` as well, ignoring now